File: 57/1.py
from typing import List

class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []
        for interval in intervals:
            if ((newInterval[0] <= interval[1] and interval[1] <= newInterval[1]) 
                or (newInterval[0] <= interval[0] and interval[0] <= newInterval[1])
                or (interval[0] <= newInterval[0] and newInterval[1] <= interval[1])):
                    newInterval[0] = min(newInterval[0], interval[0])
                    newInterval[1] = max(newInterval[1], interval[1])
            else:
                res.append(interval)

        res.append(newInterval)
        res.sort()

        return res
            
def main():

    n = int(input())
    intervals = []
    for i in range(n):
        interval = list(map(int, input().strip().split()))
        intervals.append(interval)
    newInterval = list(map(int, input().strip().split()))

    s = Solution()
    print(s.insert(intervals, newInterval))

if __name__ == "__main__":
    main()

File: 56/1.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <map>
#include <set>
#include <unordered_map>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> merged;
        for(auto interval : intervals){
            if(merged.empty() or merged.back()[1] < interval[0]){
                merged.push_back(interval);
            }else{
                merged.back()[1] = max(merged.back()[1], interval[1]);
            }
        }
        return merged;
    }
};

int main() {
    int n, num1, num2;
    vector<vector<int>> intervals;

    cin >> n;
    for(int i = 0; i < n; i++){
        cin >> num1;
        cin >> num2;
        intervals.push_back({num1, num2});
    }

    Solution solution;

    vector<vector<int>> res = solution.merge(intervals);

    for(auto vec:res){
        for(auto v:vec){
            cout << v << " ";
        }
        cout << endl;
    }
    
    return 0;
}


